mlir/lib/Dialect/SCF/IR/SCF.cpp

```diff
@@ -1090,9 +1084,13 @@ struct ForOpTensorCastFolder : public OpRewritePattern<ForOp> {
         continue;
 
       // Create a new ForOp with that iter operand replaced.
+      ValueTypeCastFnTy castFn = [](OpBuilder &b, Location loc, Type type,
```

Comment:
Though if using a `function_ref` you would have to use `auto` here (assigning to the reference could be a UAF).